### PR TITLE
Add more filters to reservation endpoint

### LINKED ISSUE #185: Add more filters to reservation endpoint
- event_subject (string)
- host_name (string)
- reserver_name (string)
- resource_name (string)
- is_favorite_resource (boolean)

### PR DESCRIPTION
**Pull request by tuomas777**
Closes #185

### CODE REVIEW

**Comment by tuomas777** [comment]
Updated, added user based reservation excluding, see the commit message for details.

File: resources/api/reservation.py [hunk 1]
```diff
@@ -378,6 +379,44 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class ReservationFilterSet(django_filters.FilterSet):
```
Comment (akx) [inline resources/api/reservation.py]: This should probably derive from `django_filters.rest_framework.FilterSet` (and then I think the `DRFFilterBooleanWidget` hack wouldn't be needed)?

Comment (tuomas777) [inline resources/api/reservation.py]: Yup it probably should, good catch.

**Comment by tuomas777** [comment]
Updated, added permission to view reservation extra fields (and those are now hidden by default)

**Pull request merged**